### DISCUSSION TITLE
jsp EL 표현식과 javascript 문법 충돌

### DISCUSSION THREAD

**Discussion by noahyunjun** [KGU-CS-HOME/.github#3] [3. 이슈]
var day = '월요일';
var text = `오늘은 ${day} 입니다.`
> jsp 코드를 거치지 않으면 원하는 대로 출력이 됩니다 
>> "오늘은 월요일 입니다.

하지만 jsp엔진을 거치게 된다면, 여러 과정을 거치게 됩니다.
1. page, request, session, context 에서 "day"를 찾습니다.
2. null이라면 아무것도 출력이 되지 않습니다.
3. 값이 존재할시 toString() 값을 출력합니다.

서버에서 request.setAttribute로 day에 값을 넣어주면, js 코드가 다르게 실행됩니다.
request.setAttribute("day", "토요일");
>> "오늘은 토요일 입니다."

해결방법 -
1. var text = `오늘은 ${'${day}'} 입니다.`
2. var text = `오늘은 \${day} 입니다.`
- 1번의 경우, 바깥쪽 ${}은 서버쪽에서 처리하고, 내부 ${}를 js에서 처리하도록 합니다.
- 2번의 경우, \를 붙임으로써 js에서 인식하게 합니다.

**Comment by gabrielyoon7** [comment]
한 가지 추가 해결 방법으로는... 이 모든 일이 *.jsp 형식에 있는 <script> 태그 내에서 발생하는 문제이므로,
함수들을 외부 파일인 *.js로 모듈화를 시킨 후 해당 함수에 인자를 넘겨주는 방법도 한 가지 대안이 될 것 같습니다.
